core/src/main/java/org/bitcoinj/crypto/HDPath.java

```diff
@@ -90,6 +107,16 @@ public HDPath(boolean hasPrivateKey, List<ChildNumber> list) {
         this.unmodifiableList = Collections.unmodifiableList(list);
     }
 
+    /**
+     * Constructs a path for a public or private key.
+     *
+     * @param prefix 'M' or 'm'
+     * @param list List of children in the path
+     */
+    private HDPath(Prefix prefix, List<ChildNumber> list) {
+        this(prefix == Prefix.PRIVATE, list);
```

Comment:
Couldn't we perhaps replace the boolean `hasPrivateKey` and the corresponding constructor, now that we have a type?

Reply:
I thought about both these things and either/both could be done in a follow-up PR. For now, I decided to stick with the boolean primitive as it more memory-efficient (although I wonder if there is a use case with 1000's or more of HDPaths.)  I also decided to keep the new constructor and everything about the new type private, but we can change that when we think public access is needed.